### PR TITLE
BUG: read_excel skips single-column empty rows

### LINKED ISSUE #39808: Add back skip_blank_lines to read_excel in pandas v>1.1.4
#### Is your feature request related to a problem?

Reading Stack Overflow, there has been several posts trying to skip blank lines when reading ([post 1](https://stackoverflow.com/questions/39334543/python-pandas-read-excel-doesnt-recognize-null-cell), [post 2](https://stackoverflow.com/questions/54890013/dont-skip-blank-lines-in-pandas-read-excel), [post 3](https://stackoverflow.com/questions/59279177/read-excel-file-pd-read-excel-by-skipping-blank-rows-only), ...). The proposed solution works for pandas v<0.24 but there is no mention about this feature and trying to use it I get:
In pandas 1.1.4 and 1.2: `TypeError: read_excel() got an unexpected keyword argument 'skip_blank_lines'` and no mention in the documentation posted above.

#### Describe the solution you'd like

Pandas `read_excel` and/or `ExcelFile` should get back this argument and functionality

### PR DESCRIPTION
**Pull request by ahawryluk**
Ref #39808

- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

This is the fix proposed in the discussion of #40095

### CODE REVIEW

File: pandas/tests/io/excel/test_readers.py [hunk 1]
```diff
@@ -1185,6 +1185,13 @@ def test_no_header_with_list_index_col(self, read_ext):
         )
         tm.assert_frame_equal(expected, result)
 
+    def test_one_col_noskip_blank_line(self, read_ext):
+        file_name = "one_col_blank_line" + read_ext
```
Comment (rhshadrach) [inline pandas/tests/io/excel/test_readers.py:1189]: Add the issue number as the first line - e.g. `# GH 39808`

File: pandas/tests/io/excel/test_readers.py [hunk 2]
```diff
@@ -1185,6 +1185,13 @@ def test_no_header_with_list_index_col(self, read_ext):
...
+        data = [0.5, np.nan, 1, 2]
+        expected = DataFrame(data, columns=["numbers"])
+        result = pd.read_excel(file_name)
+        tm.assert_frame_equal(expected, result)
```
Comment (rhshadrach) [inline pandas/tests/io/excel/test_readers.py:1193]: nit: can you change to `result, expected`

**Review by rhshadrach — CHANGES_REQUESTED** [review]
Changes look good - two requests below.

File: pandas/tests/io/excel/test_readers.py (hunk 1 above)
Comment (ahawryluk) [reply pandas/tests/io/excel/test_readers.py:1190]: done

File: pandas/tests/io/excel/test_readers.py (hunk 2 above)
Comment (ahawryluk) [reply pandas/tests/io/excel/test_readers.py]: :+1:

File: pandas/io/excel/_base.py [hunk 3]
```diff
@@ -598,6 +598,7 @@ def parse(
                     skiprows=skiprows,
                     nrows=nrows,
                     na_values=na_values,
+                    skip_blank_lines=False,
```
Comment (gfyoung) [inline pandas/io/excel/_base.py:601]: Why not expose this parameter in `read_excel` ?

Comment (ahawryluk) [reply pandas/io/excel/_base.py:601]: We tried that (#40095) but the behaviour when `skip_blank_lines=True` isn't very useful---it only skips empty spreadsheet rows if the entire spreadsheet is contained in column A. The original intent of `skip_blank_lines` was to skip `\n` lines in CSV files, but not lines with empty data elements such as `""\n` or `,,,,\n`, so there isn't an equivalent feature for spreadsheets.

**Comment by lithomas1** [comment]
Not sure if this closes the original issue, as the parameter would have to be exposed in order to truly fix. 
I think modifying this function https://github.com/pandas-dev/pandas/blob/fad96e1d3145019a5d865564bb4db008a1b1e5ea/pandas/io/parsers/python_parser.py#L596 might be able to fix the issue, but it could also introduce some side effects for CSV parsing.

File: pandas/io/excel/_base.py (hunk 3 above)
Comment (gfyoung) [reply pandas/io/excel/_base.py:601]: I see.  Can make a comment about this and reference your original PR.

Comment (ahawryluk) [reply pandas/io/excel/_base.py:601]: For sure. Do you mean an inline comment in the code, or in the description of this pull request.

**Comment by ahawryluk** [comment]
@lithomas1 I'm nervous about breaking the CSV behaviour, because it seems deliberate. I found one test where an input line `,,` is supposed to produce a row of NaNs:

https://github.com/pandas-dev/pandas/blob/154026cc7104f7b58c10a1557564eccd0f71c4a2/pandas/tests/io/parser/common/test_common_basic.py#L412-L414

Maybe this behaviour should be documented better?

**Review by rhshadrach — APPROVED** [review]
lgtm; @lithomas1 - agreed that this PR does not fully close the issue as stated there, however I'm -0 on adding the skip_blank_lines argument to read_excel. read_csv needs this because you can't tell a blank line from e.g. `,,,,` from the resulting DataFrame; this does not exist for excel files and it is straightforward to find and drop blank lines with `df.dropna(how='all')`.

@ahawryluk - can you merge master

**Comment by rhshadrach** [comment]
@simonjayhawkins @jreback - This is a bit of an odd case, wanted to get some eyes to make sure it's being handled appropriately. This PR fixes the following bug (not a regression, AFAICT) on master: when reading an excel file, blank rows would be skipped if and only if a single column is being read. This PR makes it so that blank rows are never skipped.

However, the previous workaround was to supply `skip_blank_lines=False` in `**kwds`; this argument was removed in 1.1 and so is no longer available. I'm -0 on adding `*kwds` back in (see https://github.com/pandas-dev/pandas/pull/40214#pullrequestreview-605859867).

Since this is a rather localized case, my thinking is to have this be a bugfix for 1.3 and leave the regression in 1.1.x and 1.2.x unresolved.

**Comment by simonjayhawkins** [comment]
> Since this is a rather localized case, my thinking is to have this be a bugfix for 1.3 and leave the regression in 1.1.x and 1.2.x unresolved.

sgtm. does the OP need to change to not close the issue?

**Comment by ahawryluk** [comment]
@simonjayhawkins, thanks for taking a look at this. My preference is to close #39808 based on

- the OP of https://github.com/pandas-dev/pandas/issues/39808#issuecomment-784366880 prefers the current solution now that we understand how `skip_blank_lines` really works
- IIUC, @rhshadrach and I don't think `skip_blank_lines` should be added to `read_excel` in the future

But I'm happy to edit the PR if everyone prefers to leave #39808 open.

**Comment by simonjayhawkins** [comment]
> * IIUC, @rhshadrach and I don't think `skip_blank_lines` should be added to `read_excel` in the future

The issue could be closed once that is agreed as they way forward on the issue itself.

**Review by jreback — APPROVED** [review]
lgtm. back to you @rhshadrach

**Comment by jreback** [comment]
for 1.3

**Comment by rhshadrach** [comment]
Thanks @ahawryluk! I've left the issue open for now, but will put my thoughts there shortly.

**Pull request merged**